packages/webapp/src/containers/SensorReadings/saga.js

```diff
@@ -43,157 +43,193 @@ import i18n from '../../locales/i18n';
+    const data = result?.data;
+    let sensorDataByLocationIds = {};
 
-      if (readingType === TEMPERATURE) {
-        params = {
-          ...params,
-          lat: centerPoint?.lat ?? lat,
-          lon: centerPoint?.lng ?? lng,
-        };
-        const openWeatherPromiseList = [];
-        for (const weatherURL of OPEN_WEATHER_API_URL_FOR_SENSORS) {
-          const openWeatherUrl = new URL(weatherURL);
-          for (const key in params) {
-            openWeatherUrl.searchParams.append(key, params[key]);
-          }
-          if (weatherURL === DAILY_FORECAST_API_URL) {
-            openWeatherUrl.searchParams.append('cnt', 1);
-          }
-          openWeatherPromiseList.push(call(axios.get, openWeatherUrl?.toString()));
-        }
+    for (let locationId of locationIds) {
+      sensorDataByLocationIds[locationId] = {};
+      for (let type of readingTypes) {
+        sensorDataByLocationIds[locationId][type] = {};
 
-        const [openWeatherResponse, predictedWeatherResponse, predictedDailyWeatherResponse] =
-          yield all(openWeatherPromiseList);
-        stationName = predictedDailyWeatherResponse?.data?.city.name;
-        const weatherResData = [
-          ...openWeatherResponse.data.list,
-          ...predictedWeatherResponse.data.list,
-        ];
+        let readings = sensorDataByLocationIds[locationId][type];
+        readings.selectedSensorName = data?.sensorsPoints[0]?.name;
+        readings.lastUpdatedReadingsTime = getLastUpdatedTime(
+          data?.sensorReading[type]
+            .filter((cv) => (cv.value ? cv.value : cv.value === 0))
+            .map((cv) => new Date(cv.actual_read_time).valueOf() / 1000),
+        );
+        readings.predictedXAxisLabel = roundDownToNearestChosenPoint(currentUnixTime);
 
-        const ambientData = weatherResData.reduce((acc, tempInfo) => {
-          let dateAndTimeInfo = new Date(tempInfo?.dt * 1000).toString();
-          const isCorrectTimestamp = GRAPH_TIMESTAMPS?.find((g) => dateAndTimeInfo?.includes(g));
-          if (isCorrectTimestamp && startDate < tempInfo?.dt && tempInfo?.dt < predictedEndDate) {
-            const currentDateTime = `${dateAndTimeInfo?.split(':00:00')[0]}:00`;
-            if (!isFound && currentDT < tempInfo?.dt) {
-              isFound = true;
-              predictedXAxisLabel = acc[Object.keys(acc).at(-1)][CURRENT_DATE_TIME];
-            }
-            if (!acc[tempInfo?.dt]) acc[tempInfo?.dt] = {};
-            acc[tempInfo?.dt] = {
-              ...acc[tempInfo?.dt],
-              [`${ambientTempFor} ${stationName}`]: tempInfo?.main?.temp,
+        // reduce sensor data
+        let typeReadings = {};
+        typeReadings = data?.sensorReading[type].reduce((acc, cv) => {
+          const currentValueUnixDate = new Date(cv?.read_time).getTime() / 1000;
+          const currentValueUnixTime = new Date(currentValueUnixDate * 1000).toString();
```

Comment:
currentValueUnixTimeMsString done lol

Reply:
This one isn't a ms string at all though! It's a Date-Time string (see above).

The ms (1000 multiplier) was just to convert it so that native JavaScript could parse it.